### DISCUSSION TITLE
oo/native-method

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#257] [Announcements]
# oo/native-method

掌握C语言实现Java native方法的技巧，提升Java程序性能，深入理解Java Native Interface (JNI)。跟随我们的手把手教程，轻松实现跨语言编程。

https://javabetter.cn/oo/native-method.html

**Comment by ustcjsj** [comment]
linux环境下，编译运行的命令：
#编译
gcc -I"$JAVA_HOME/include" -I"$JAVA_HOME/include/linux" -fPIC -shared -o libhello.so HelloJNI.c
#运行（指定动态库路径）
java -Djava.library.path=. HelloJNI

**Comment by lcicl** [comment]
PS：Java 9 以后，javah 被弃用，取而代之的是使用 -h 选项来生成头文件，例如 javac -h . ClassName.java。

这里应该是javac -h  -jni .ClassName.java。
